packages/cli/lib/configs.js

```diff
@@ -0,0 +1,53 @@
+const request = require('request-promise-native');
+const { logger } = require('@hubspot/cli-lib/logger');
+const { logErrorInstance } = require('@hubspot/cli-lib/errorHandlers');
+
+// https://developer.github.com/v3/#user-agent-required
+const USER_AGENT_HEADERS = { 'User-Agent': 'HubSpot/hubspot-cms-tools' };
+
+/**
+ * @param {String} repoName - name of the repository
+ * @param {String} filePath - path where config file is stored
+ */
+async function fetchRawConfig(repoName, filePath) {
+  const URI = `https://api.github.com/repos/HubSpot/${repoName}/contents/${filePath}`;
+  try {
+    return await request.get(URI, {
+      headers: { ...USER_AGENT_HEADERS },
+      json: true,
+    });
+  } catch (err) {
+    if (err.statusCode === 404) {
+      logger.error(`Config file not found.`);
+    }
+  }
+  return null;
+}
+
+/**
+ * @param {String} filePath - path where config file is stored
+ * @param {String} repoName - name of the github repository
+ * @returns {Buffer|Null} Zip data buffer
+ */
+async function downloadConfig(repoName, filePath) {
```

Comment:
While this works, it does seem like a pretty indirect way to get a JSON file with information on the different samples.

Reply:
Simplified the logic of getting configs to 1 method, and made it more abstract for getting json files.